src/Samples/SarifToCsv/SdkExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.CodeAnalysis.Sarif;
+using System.Collections.Generic;
+
+namespace SarifToCsv
+{
+    public static class SdkExtensions
+    {
+        public static IEnumerable<PhysicalLocation> PhysicalLocations(this Result result)
+        {
+            foreach (Location location in result.Locations)
+            {
+                if (location.PhysicalLocation != null)
+                {
+                    yield return location.PhysicalLocation;
+                }
+            }
+        }
+
+        public static string FileUri(this FileLocation fileLocation, Run run)
+        {
+            if (fileLocation == null)
+            {
+                return null;
+            }
+            else if (fileLocation.Uri != null)
```

Comment:
>else if [](start = 12, length = 7)

you don't required chained ifs here since each conditional clause returns. would make more concise to eliminate them (as well as final else clause.

Reply:
I intended this pattern to help readability; the if.. else if... else if meaning to show we're only taking one of these paths.